java/src/main/java/org/apache/lucene/analysis/hebrew/HebrewQueryLightAnalyzer.java

```diff
@@ -39,40 +44,17 @@ public HebrewQueryLightAnalyzer() throws IOException {
     protected TokenStreamComponents createComponents(final String fieldName, final Reader reader) {
         // on query - if marked as keyword don't keep origin, else only lemmatized (don't suffix)
         // if word termintates with $ will output word$, else will output all lemmas or word$ if OOV
-        final StreamLemmasFilter src = new StreamLemmasFilter(reader, dict, SPECIAL_TOKENIZATION_CASES, commonWords, lemmaFilter);
-        src.setKeepOriginalWord(false);
+        // TODO: use special tokenization cases
+        HebrewTokenizer src = new HebrewTokenizer(reader, dict.getPref());
         src.setSuffixForExactMatch(originalTermSuffix);
-
-        TokenStream tok = new ASCIIFoldingFilter(src);
-        //tok = new SuffixKeywordFilter(tok, '$');
-        tok = new AddSuffixFilter(tok, '$') {
-            @Override
-            protected void handleCurrentToken() {
-                if (HebrewTokenizer.tokenTypeSignature(HebrewTokenizer.TOKEN_TYPES.Hebrew).equals(typeAtt.type())) {
-                    if (keywordAtt.isKeyword())
-                        suffixCurrent();
-                    return;
-                }
-
-                if (HebrewTokenizer.tokenTypeSignature(HebrewTokenizer.TOKEN_TYPES.NonHebrew).equals(typeAtt.type())) {
-                    if (keywordAtt.isKeyword()) {
-                        suffixCurrent();
-                        return;
-                    }
-                }
-
-                if (CommonGramsFilter.GRAM_TYPE.equals(typeAtt.type()) ||
-                        HebrewTokenizer.tokenTypeSignature(HebrewTokenizer.TOKEN_TYPES.Numeric).equals(typeAtt.type()) ||
-                        HebrewTokenizer.tokenTypeSignature(HebrewTokenizer.TOKEN_TYPES.Mixed).equals(typeAtt.type())) {
-                    keywordAtt.setKeyword(true);
-                    return;
-                }
-
-                duplicateCurrentToken();
-                suffixCurrent();
-            }
-        };
+        TokenStream tok = new NiqqudFilter(src);
+        tok = new ASCIIFoldingFilter(tok);
+        tok = new LowerCaseFilter(tok);
+        tok = new HebrewLemmatizerTokenFilter(tok, dict, false, false);
+        tok = new IgnoreOriginalTokenFilter(tok);
+        tok = new AddSuffixTokenFilter(tok, '$');
```

Comment:
Not my point, some tokens need to be duplicated (i.e. in IndexingAnalyzer ים is going to be indexed both as ים and ים$, at the same index poisition). Your change prevents that from happening.